src/StaticWebAssetsSdk/Tasks/ResolveCompressedAssets.cs

```diff
@@ -0,0 +1,256 @@
+
+                if (!existingCompressionFormatsByAssetItemSpec.TryGetValue(relatedAssetItemSpec, out var existingFormats))
+                {
+                    existingFormats = new();
+                    existingCompressionFormatsByAssetItemSpec.Add(relatedAssetItemSpec, existingFormats);
+                }
+
+                var compressionFormat = asset.GetMetadata("AssetTraitValue");
+
+                if (!CompressionConfiguration.CompressionFormat.IsValidCompressionFormat(compressionFormat))
+                {
+                    Log.LogError(
+                        "The asset '{0}' has an unknown compression format '{1}'.",
+                        asset.ItemSpec,
+                        compressionFormat);
+                }
+
+                existingFormats.Add(compressionFormat);
+            }
+        }
+
+        // Generate internal representations of each compression configuration.
+        var compressionConfigurations = CompressionConfigurations
+            .Select(CompressionConfiguration.FromTaskItem)
+            .ToArray();
+        var candidateAssetsByConfigurationName = compressionConfigurations
+            .ToDictionary(cc => cc.ItemSpec, _ => new List<ITaskItem>());
+
+        // Add each explicitly-provided asset as a candidate asset for its specified compression configuration.
+        if (ExplicitAssets is not null)
+        {
+            foreach (var asset in ExplicitAssets)
+            {
+                var configurationName = asset.GetMetadata("ConfigurationName");
+                if (candidateAssetsByConfigurationName.TryGetValue(configurationName, out var candidateAssets))
+                {
+                    candidateAssets.Add(asset);
+                    Log.LogMessage(
+                        "Explicitly-specified compressed asset '{0}' matches known compression configuration '{1}'.",
+                        asset.ItemSpec,
+                        configurationName);
+                }
+                else
+                {
+                    Log.LogError(
+                        "Explicitly-specified compressed asset '{0}' has an unknown compression configuration '{1}'.",
+                        asset.ItemSpec,
+                        configurationName);
+                }
+            }
+        }
+
+        // Add each candidate asset to each compression configuration with a matching pattern.
+        foreach (var asset in CandidateAssets)
+        {
+            if (IsCompressedAsset(asset))
+            {
+                Log.LogMessage(
+                    MessageImportance.Low,
+                    "Ignoring asset '{0}' for compression because it is already compressed.",
+                    asset.ItemSpec);
+                continue;
+            }
+
+            var relativePath = asset.GetMetadata("RelativePath");
+
+            foreach (var configuration in compressionConfigurations)
+            {
+                if (configuration.Matches(relativePath))
```

Comment:
All the include/excludes should be evaluated against all the assets (or we should have a global set of `excludes`) that can be defined globally.

We want to avoid situations where if we said, never to compress a format (for example .woff) we accidentally include it because a separate pattern was too loose and included it.

Reply:
Sounds good. I've added a `GlobalCompressionExcludePattern` MSBuild property to decide which exclude patterns are "global".